typeql/language/rule-validation.feature

```diff
@@ -394,6 +394,25 @@ Feature: TypeQL Rule Validation
       """
 
 
+  Scenario: variable declared in when cannot be redeclared in then
+    Then typeql define; throws exception
+      """
+      define
+      person sub entity;
+      parentship sub relation, relates father, relates son;
+      person plays parentship:father;
+      person plays parentship:son;
+      rule my-rule-label:
+          when {
+              $s isa person
+              $f isa person
+              $relation(father:$f, son:$s) isa parentship; $brother isa person;
+          } then {
+          $relation(father:$f, son:$brother) isa parentship;
+          };
+      """
+
+
```

Comment:
This still seems out of place within the file, given there's a section of rules for errors concerning variables in the when and then